File: PYTHON/Program/Komputer.py
from CPU import CPU
from Ram import Ram
from Harddrive import Harddrive
from OutputDevice import OutputDevice
from InputDevice import InputDevice
from Monitor import Monitor
from Komponen import Komponen

class Komputer:
    def __init__(self, nama="", cpu=CPU(), ramList=[], harddrive=Harddrive(), outputDeviceList=[], inputDeviceList=[], monitorList=[]):
        self.nama = nama
        self.cpu = cpu
        self.ramList = ramList
        self.harddrive = harddrive
        self.outputDeviceList = outputDeviceList
        self.inputDeviceList = inputDeviceList
        self.monitorList = monitorList

    def setNama(self, nama):
        self.nama = nama

    def setCpu(self, cpu):
        self.cpu = cpu

    def setRam(self, ramList):
        self.ramList = ramList

    def setHarddrive(self, harddrive):
        self.harddrive = harddrive

    def addRam(self, ram):
        self.ramList.append(ram)

    def getNama(self):
        return self.nama

    def getCpu(self):
        return self.cpu

    def getRamList(self):
        return self.ramList

    def getHarddrive(self):
        return self.harddrive

    def getOutputDeviceList(self):
        return self.outputDeviceList

    def getInputDeviceList(self):
        return self.inputDeviceList

    def getMonitorList(self):
        return self.monitorList

    def addOutputDevice(self, outputDevice):
        self.outputDeviceList.append(outputDevice)

    def addInputDevice(self, inputDevice):
        self.inputDeviceList.append(inputDevice)

    def addMonitor(self, monitor):
        self.monitorList.append(monitor)

    def setOutputdevice(self, outputDeviceList):
        self.outputDeviceList = outputDeviceList

    def setInputdevice(self, inputDeviceList):
        self.inputDeviceList = inputDeviceList

    def setMonitor(self, monitorList):
        self.monitorList = monitorList





# #pragma once
# #include <iostream>
# #include <string>
# #include <vector>
# #include "Cpu.cpp"
# #include "Harddrive.cpp"
# #include "Ram.cpp"
# #include "OutputDevice.cpp"
# #include "InputDevice.cpp"
# #include "Monitor.cpp"

# using namespace std;


# class Komputer
# {
# private:
#     string nama;
#     CPU cpu;
#     vector<Ram> ramList;
#     Harddrive harddrive;
#     vector<OutputDevice> outputDeviceList;
#     vector<InputDevice> inputDeviceList;
#     vector<Monitor> monitorList;
    
# public:
#     Komputer()
#     {


#     }


#     Komputer(string nama, CPU cpu, vector<Ram> ramList, Harddrive harddrive, vector<OutputDevice> outputDeviceList, vector<InputDevice> inputDeviceList, vector<Monitor> monitorList)
#     {
#         this->nama = nama;
#         this->cpu = cpu;
#         this->ramList = ramList;
#         this->harddrive = harddrive;
#         this->outputDeviceList = outputDeviceList;
#         this->inputDeviceList = inputDeviceList;
#         this->monitorList = monitorList;
#     }


#     void setNama(string nama)
#     {
#         this->nama = nama;
#     }

#     void setCpu(CPU cpu)
#     {
#         this->cpu = cpu;
#     }


#     void setRam(vector<Ram> ramList)
#     {
#         this->ramList = ramList;
#     }    
   
#     void setHarddrive(Harddrive harddrive)
#     {
#         this->harddrive = harddrive;
#     }


#     void addRam(Ram ram)
#     {
#         this->ramList.push_back(ram);
#     }


#     string getNama()
#     {
#         return this->nama;
#     }


#     CPU getCpu()
#     {
#         return this->cpu;
#     }


#     vector<Ram> getRamList() // return vector
#     {
#         return this->ramList;
#     }


#     Harddrive getHarddrive()
#     {
#         return this->harddrive;
#     }
#     vector<OutputDevice> getOutputDeviceList()
#     {
#         return this->outputDeviceList;
#     }
#     vector<InputDevice> getInputDeviceList()
#     {
#         return this->inputDeviceList;
#     }
#     vector<Monitor> getMonitorList()
#     {
#         return this->monitorList;
#     }
#     void addOutputDevice(OutputDevice outputDevice)
#     {
#         this->outputDeviceList.push_back(outputDevice);
#     }
#     void addInputDevice(InputDevice inputDevice)
#     {
#         this->inputDeviceList.push_back(inputDevice);
#     }
#     void addMonitor(Monitor monitor)
#     {
#         this->monitorList.push_back(monitor);
#     }

#     void setOutputdevice(vector<OutputDevice> outputDeviceList)
#     {
#         this->outputDeviceList = outputDeviceList;
#     }
#     void setInputdevice(vector<InputDevice> inputDeviceList)
#     {
#         this->inputDeviceList = inputDeviceList;
#     }
#     void setMonitor(vector<Monitor> monitorList)
#     {
#         this->monitorList = monitorList;
#     }
    


#     ~Komputer()
#     {


#     }
# };

File: JAVA/Program/Komputer.java
package Program;
import java.util.List;

public class Komputer {
    private String nama;
    private CPU cpu;
    private List<Ram> ramList;
    private Harddrive harddrive;
    private List<OutputDevice> outputDeviceList;
    private List<InputDevice> inputDeviceList;
    private List<Monitor> monitorList;

    public Komputer() {
    }

    public Komputer(String nama, CPU cpu, List<Ram> ramList, Harddrive harddrive, List<OutputDevice> outputDeviceList, List<InputDevice> inputDeviceList, List<Monitor> monitorList) {
        this.nama = nama;
        this.cpu = cpu;
        this.ramList = ramList;
        this.harddrive = harddrive;
        this.outputDeviceList = outputDeviceList;
        this.inputDeviceList = inputDeviceList;
        this.monitorList = monitorList;
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    public void setCpu(CPU cpu) {
        this.cpu = cpu;
    }

    public void setRam(List<Ram> ramList) {
        this.ramList = ramList;
    }

    public void setHarddrive(Harddrive harddrive) {
        this.harddrive = harddrive;
    }

    public void addRam(Ram ram) {
        this.ramList.add(ram);
    }

    public String getNama() {
        return this.nama;
    }

    public CPU getCpu() {
        return this.cpu;
    }

    public List<Ram> getRamList() {
        return this.ramList;
    }

    public Harddrive getHarddrive() {
        return this.harddrive;
    }

    public List<OutputDevice> getOutputDeviceList() {
        return this.outputDeviceList;
    }

    public List<InputDevice> getInputDeviceList() {
        return this.inputDeviceList;
    }

    public List<Monitor> getMonitorList() {
        return this.monitorList;
    }

    public void addOutputDevice(OutputDevice outputDevice) {
        this.outputDeviceList.add(outputDevice);
    }

    public void addInputDevice(InputDevice inputDevice) {
        this.inputDeviceList.add(inputDevice);
    }

    public void addMonitor(Monitor monitor) {
        this.monitorList.add(monitor);
    }

    public void setOutputDeviceList(List<OutputDevice> outputDeviceList) {
        this.outputDeviceList = outputDeviceList;
    }

    public void setInputDeviceList(List<InputDevice> inputDeviceList) {
        this.inputDeviceList = inputDeviceList;
    }

    public void setMonitorList(List<Monitor> monitorList) {
        this.monitorList = monitorList;
    }
}
